test/tools/catch-analyzer/src/main/java/org/elasticsearch/test/ClassAnalyzer.java

```
@@ -0,0 +1,172 @@
+        analyses.put(new Method(name, desc), analyzer);
+        return analyzer;
+    }
+
+    @Override
+    public void visitEnd() {
+        // fold lambdas violations into their parent methods
+        List<Method> lambdas = new ArrayList<>();
+        analyses.forEach((method, analysis) -> {
+            for (Method lambda : analysis.lambdas) {
+                MethodAnalyzer lambdaAnalysis = analyses.get(lambda);
+                if (lambdaAnalysis != null) {
+                    analysis.violations.addAll(lambdaAnalysis.violations);
+                    lambdas.add(lambda);
+                } else {
+                    throw new AssertionError("where in the world is lambda " + lambda);
+                }
+            }
+        });
+        // now nuke them
+        analyses.keySet().removeAll(lambdas);
+    }
+    
+    /** log violations, grouping them by catch block entry line number for the method */
+    long logViolations(PrintStream stream) {
+        long violationCount = 0;
+        if (suppressed) {
+            long violations = 0;
+            for (MethodAnalyzer analysis : analyses.values()) {
+                violations += analysis.violations.size();
+            }
+            if (violations == 0) {
+                StringBuilder sb = createMessage("Wrong class annotation", null, -1);
+                addDetails(sb, new Violation(Kind.ANNOTATED_AS_SWALLOWER_BUT_HAS_NO_PROBLEMS, -1, -1));
+                stream.println(sb.toString());
+                violationCount++;
+            }
+        } else {
+            for (Map.Entry<Method,MethodAnalyzer> entry : analyses.entrySet()) {
+                Method method = entry.getKey();
+                MethodAnalyzer analysis = entry.getValue();
+                if (analysis.suppressed) {
+                    if (analysis.violations.isEmpty()) {
+                        StringBuilder sb = createMessage("Wrong method annotation", method, -1);
```

Should this string be "Method marked as `@SwallowsException` but doesn't swallow any exceptions" or am reading this wrong?